File: backend/src/main/scala/domains/notification/objects/request/CreateNotificationRequest.scala
package domains.notification.objects.request

import domains.notification.objects.{NotificationKind, NotificationPayload}
import domains.user.objects.Username
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

/** 内部创建通知请求体，包含收件人、触发人、文案 key、payload 和跳转目标。 */
final case class CreateNotificationRequest(
  recipientUsername: Username,
  actorUsername: Option[Username],
  kind: NotificationKind,
  titleKey: String,
  bodyKey: String,
  payload: NotificationPayload,
  targetPath: String,
  targetAnchor: Option[String]
)

/** 提供创建通知请求体 JSON codec。 */
object CreateNotificationRequest:
  given Encoder[CreateNotificationRequest] = deriveEncoder[CreateNotificationRequest]
  given Decoder[CreateNotificationRequest] = deriveDecoder[CreateNotificationRequest]

File: frontend/src/objects/notification/request/CreateNotificationRequest.ts
import type { NotificationKind } from '@/objects/notification/NotificationKind'
import type { NotificationPayload } from '@/objects/notification/NotificationPayload'
import type { Username } from '@/objects/user/Username'

/** 内部创建通知请求体；title/body key 和 payload 由调用方按通知类型配套提供。 */
export type CreateNotificationRequest = {
  recipientUsername: Username
  actorUsername: Username | null
  kind: NotificationKind
  titleKey: string
  bodyKey: string
  payload: NotificationPayload
  targetPath: string
  targetAnchor: string | null
}
